Description: a set of changes from git://github.com/mariuz/perl-dbd-interbase.git
Author: Popa Marius Adrian <http://mapopa.blogspot.com>
Origin: git://github.com/mariuz/perl-dbd-interbase.git

diff --git a/Makefile.PL b/Makefile.PL
index 70c783b..a000a94 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -379,6 +379,38 @@ LDDLFLAGS = /LIBPATH:\"$vc_lib\" $cur_lddlflags
     '
     } ";
     }
+    # Support for MinGW ??? :)
+    elsif ($Config::Config{'cc'} eq "gcc") {
+        print "MinGW gcc found?\n";
+        my $MGW_PATH = 'C:/mystrawberry/c';
+
+        # Copy from above
+        my $mingw_inc = $MGW_PATH . "/include";
+        my $mingw_lib = $MGW_PATH . "/lib";
+
+        $INC .= " -I\"$mingw_inc\"";
+
+        my $ib_lib = dir_choice("InterBase lib directory",
+                            [$ib_lib_dir . "SDK\\lib_ms", $ib_lib_dir . "lib"],
+                            [qw(gds32_ms.lib fbclient_ms.lib)]);
+
+        my $cur_libs = $Config::Config{'libs'} ;
+        my $cur_lddlflags = $Config::Config{'lddlflags'} ;
+
+        my $lib;
+        if (-f "$ib_lib/fbclient_ms.lib")
+            { $lib = "$ib_lib/fbclient_ms.lib"; }
+     else
+            { $lib = "$ib_lib/gds32_ms.lib"; }
+
+        eval "
+    sub MY::const_loadlibs {
+    '
+LDLOADLIBS = \"$lib\" $cur_libs
+LDDLFLAGS = /LIBPATH:\"$mingw_lib\" $cur_lddlflags
+    '
+    } ";
+    }
     else
     {
         # Borland C++ 5.5
diff --git a/PatchBCC b/PatchBCC
index 7644573..ae48d39 100755
--- a/PatchBCC
+++ b/PatchBCC
@@ -144,4 +144,4 @@ if (! -e $coffname)
 
 ################################################################################
 # EOF
-################################################################################
\ No newline at end of file
+################################################################################
diff --git a/README b/README
index 566544c..2443c43 100644
--- a/README
+++ b/README
@@ -15,23 +15,14 @@ DBD::InterBase -- DBI driver for Firebird and InterBase RDBMS server.
 REQUIREMENTS:
   - Perl 5.004 installed
   - Perl DBI (1.41 and higher)
-  - Firebird or InterBase
+  - Firebird (2.1 and higher)
   - A C compiler
     * UN*X
       GCC 2.95.2 (tested, older may or may not work) or other capable
 
     * Windows
-      - Borland C++ 5.5
-        You can donwload a free copy of BCC55 at
-          ftp://ftpd.borland.com/download/bcppbuilder/freecommandLinetools.exe
-
-        In order to compile with Borland compiler you need pmake:
-          http://www.cpan.org/authors/id/NI-S/Make-1.00.tar.gz
-        dmake would work too (building) but has problems creating PPD file
-          http://www.cpan.org/authors/id/GSAR/dmake-4.1pl1-win32.zip
-
-
-      - or Microsoft VC++ 6.0
+      - Strawberry perl comes with it's own compiler (mingw) 
+      - or Microsoft Express VC++ 9.0 (2008)
 
 
 *BEFORE* BUILDING, TESTING AND INSTALLING this you will need to:
@@ -44,18 +35,16 @@ REQUIREMENTS:
 
   - Remember to *read* the DBI README file!
 
-  - Make sure that InterBase/Firebird server is running (for testing)
+  - Make sure that Firebird server is running (for testing telnet localhost 3050)
 
 
 BUILDING:
-  Win32 with Borland C++:
-    type 'pmake' or 'dmake', not just 'make'. before run make: 
-      perl PatchBCC
+  Win32/Win64 with Strawberry 
+    type 'make' from the console 
 
-  Win32 with MS compiler:
+  Win32/Win64 with MS compiler:
     type 'nmake', not just 'make'
 
-
   To Configure and build the DBD:
     perl Makefile.PL
       (when asked to specify a test database, please specify a non-existing file)
diff --git a/dbdimp.c b/dbdimp.c
index 932a103..e90e9ab 100644
--- a/dbdimp.c
+++ b/dbdimp.c
@@ -892,6 +892,7 @@ int dbd_st_prepare(SV *sth, imp_sth_t *imp_sth, char *statement, SV *attribs)
             break;
 
         case isc_info_sql_stmt_ddl:
+	case isc_info_sql_stmt_set_generator:
         case isc_info_sql_stmt_exec_procedure:
             /* no count_item to gather */
             break;
